x/mongo/driver/topology/server.go

```diff
@@ -288,7 +294,7 @@ func (s *Server) ProcessHandshakeError(err error, startingGenerationNumber uint6
 	// the description.Server appropriately. The description should not have a TopologyVersion because the staleness
 	// checking logic above has already determined that this description is not stale.
 	s.updateDescription(description.NewServerFromError(s.address, wrappedConnErr, nil))
```

Comment:
To check, would updating the server description to unknown make it ineligible for server selection (and similarly in `ProcessError`?

Reply:
The spec actually prohibits us from marking the server Unknown to ensure that the LB is always selectable. As part of a previous PR in this epic, I changed [updateDescription](https://github.com/mongodb/mongo-go-driver/blob/31673c03aaee62e2ef8402883e23d9eabb0b6280/x/mongo/driver/topology/server.go#L552) to be a no-op when the server type is `LoadBalancer`. I felt this was a cleaner approach than checking every invocation.

Reply:
Ah right, that makes sense. Thanks for the explanations!